llvm/lib/Target/AMDGPU/SIMemoryLegalizer.cpp

```diff
@@ -2670,12 +2754,25 @@ bool SIMemoryLegalizer::expandAtomicCmpxchgOrRmw(const SIMemOpInfo &MOI,
   return Changed;
 }
 
+static bool isModuleUsingVulkanMM(MachineFunction &MF) {
+  Module *M = MF.getFunction().getParent();
+  if (auto *MD = mdconst::extract_or_null<ConstantInt>(
+          M->getModuleFlag("amdgpu.vulkan.memory-model")))
+    return MD->getZExtValue() == 1;
+  return false;
+}
+
```

Comment:
This is worse than the chicken bit cl::opt. Moving into the IR makes the IR modal and this more permanent

Reply:
It's actually not that bad. `VulkanMemoryModel` is a "capability" that must be declared by a SPIRV module if it uses the `NonPrivatePointer` feature. It is reasonable for the compiler to translate this to a module flag in LLVM IR.